server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/stats/AdmissionControlStats.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ratelimitting.admissioncontrol.stats;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Class for admission control stats used as part of node stats
+ * @opensearch.internal
+ */
+public class AdmissionControlStats implements ToXContentFragment, Writeable {
+
+    private final List<AdmissionControllerStats> admissionControllerStatsList;
+
+    /**
+     *
+     * @param admissionControllerStatsList list of admissionControllerStats
+     */
+    public AdmissionControlStats(List<AdmissionControllerStats> admissionControllerStatsList) {
+        this.admissionControllerStatsList = admissionControllerStatsList;
+    }
+
+    /**
+     *
+     * @param in the stream to read from
+     * @throws IOException if an I/O error occurs
+     */
+    public AdmissionControlStats(StreamInput in) throws IOException {
+        this.admissionControllerStatsList = in.readList(AdmissionControllerStats::new);
+    }
+
+    /**
+     * Write this into the {@linkplain StreamOutput}.
+     *
+     * @param out the output stream to write entity content to
+     */
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeList(this.admissionControllerStatsList);
+    }
+
+    public List<AdmissionControllerStats> getAdmissionControllerStatsList() {
+        return admissionControllerStatsList;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject("admission_control");
+        this.admissionControllerStatsList.forEach(stats -> {
+            try {
+                builder.field(stats.getAdmissionControllerName(), stats);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+        return builder.endObject();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) return false;
+        AdmissionControlStats admissionControlStats = (AdmissionControlStats) obj;
+        return this.getAdmissionControllerStatsList().size() == admissionControlStats.getAdmissionControllerStatsList().size();
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
+    }
```

Comment:
why super?

Reply:
We don't need explicit `equals` and `hashcode` methods implementation. It was removed in updated PR. There is no explicit usage for these methods it was added to evaluate serialization validation in tests and now it was evaluated individually.